src/lib.rs

```diff
@@ -148,6 +157,39 @@ impl Game {
             if inputs.right {
                 player.x += self.speed * time_steps_passed
             }
+
+            if !inputs.q && self.last_input.q {
+                kill_position = Some((player.x, player.y));
+            }
+        }
+
+        match kill_position {
+            Some(position) => self.kill_player_near(position)?,
+            None => return Ok(()),
+        }
+
+        Ok(())
+    }
+
+    fn kill_player_near(&mut self, position: (f64, f64)) -> Result<(), &'static str> {
+        let local_player_color = match &self.local_player_color {
+            None => return Ok(()), // not controlling anything
+            Some(c) => *c,
+        };
+
+        for player in self.players.iter_mut() {
+            if player.color == local_player_color {
+                continue;
+            }
+
+            let distance = ((position.0 - player.x).powi(2) + (position.1 - player.y).powi(2))
+                .sqrt()
+                .abs();
+
+            if distance < self.kill_distance {
```

Comment:
This kills _a_ nearby player. Would be better to kill the _nearest_ player I think. TODO?

Reply:
Yeah, good point. I figured the "rules" of killing are generally TBD, but can do whatever.

Reply:
For the time being it's a battle royal!